src/netbase.cpp

```diff
@@ -668,8 +669,8 @@ static bool ConnectThroughProxy(const proxyType &proxy,
     if (proxy.randomize_credentials)
     {
         ProxyCredentials random_auth;
-        random_auth.username = strprintf("%i", insecure_rand());
-        random_auth.password = strprintf("%i", insecure_rand());
+        static std::atomic_int counter;
```

Comment:
initialize to a random value

Reply:
this change has been already merged via #1005, but I will initialize the counter to a random value.